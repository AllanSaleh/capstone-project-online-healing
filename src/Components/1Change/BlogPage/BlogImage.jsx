import React from 'react';

export default function BlogImage({ image }) {
  return (
    <div className="flex flex-col justify-evenly h-firstsection px-sides pt-navbar">
      <img className="w-full" src={image} alt="Blog" />
    </div>
  );
}
