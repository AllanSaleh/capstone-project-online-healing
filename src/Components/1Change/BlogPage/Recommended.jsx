import React from 'react';

import Blog1 from './Images/Blog1.svg';
import Send from '../Footer/Images/Send.svg';
import Blog2 from './Images/Blog2.svg';

export default function Recommended() {
  return (
    <div className="px-sides h-section flex flex-col justify-around">
      <div className="flex flex-col justify-between items-start h-40">
        <div>
          <div className="text-title leading-tight">Sign Up For The Healing Blog</div>
          <div className="text-paragraph text-gray-700">
            Weekly, Ad-Free Blogs That Help You Stay In The Know!
          </div>
        </div>

        <div className="flex  border-2 border-gray-600 h-12 rounded-lg">
          <input className="w-64 p-2 rounded-l-md" placeholder="Enter your E-Mail" type="email" />
          <button
            className="border-0 border-l-2 border-gray-600 p-2 bg-blue-dark rounded-r-md"
            type="button"
          >
            <img src={Send} alt="Send" />
          </button>
        </div>
      </div>

      <div>
        <div className="uppercase text-subtitle mb-8">recommended for you</div>

        <div className="flex justify-between">
          <img className="w-full mr-8" src={Blog2} alt="blog2" />
          <img className="w-full" src={Blog1} alt="blog1" />
        </div>
      </div>
    </div>
  );
}
