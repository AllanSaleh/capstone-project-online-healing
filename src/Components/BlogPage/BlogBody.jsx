import React from 'react';

export default function BlogBody({ title, intro, subtitle, body }) {
  return (
    <div className="flex flex-col justify-evenly lg:h-section px-sides my-4">
      <div className="text-3xl lg:text-title">{title}</div>
      <div className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify">
        "{intro}"
      </div>
      <div className="mt-4 lg:mt-0 text-xl lg:text-subtitle">{subtitle}</div>
      <div className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify">
        {body.map((paragraph) => (
          <>
            <p>{paragraph}</p>
            <br />
          </>
        ))}
      </div>
    </div>
  );
}
