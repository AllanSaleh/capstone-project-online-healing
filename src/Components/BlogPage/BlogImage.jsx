import React from 'react';

export default function BlogImage({ image }) {
  return (
    <div className="flex items-center lg:h-firstsection px-sides pt-24 lg:pt-32">
      <img className="h-full" src={image} alt="Blog" />
    </div>
  );
}
