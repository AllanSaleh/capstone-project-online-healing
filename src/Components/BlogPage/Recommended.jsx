import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

import Blog1 from './Images/Blog1.svg';
import Send from '../Footer/Images/Send.svg';
import Blog2 from './Images/Blog2.svg';

export default function Recommended() {
  const history = useHistory();

  const [email, setEmail] = useState('');

  const subscribe = () => {
    if (email.match(/[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9]+[.]+[a-zA-Z0-9]/))
      history.push({
        pathname: '/ThankYou',
        state: 'You email has been added to the mailing list successfully!',
      });
    else alert('Please enter a valid E-Mail!');
  };

  return (
    <div className="px-sides lg:h-section flex flex-col justify-around">
      <div className="flex flex-col justify-between items-start lg:h-40">
        <div>
          <div className="text-3xl lg:text-title leading-tight">Sign Up For The Healing Blog</div>
          <div className="text-lg lg:text-paragraph text-gray-700">
            Weekly, Ad-Free Blogs That Help You Stay In The Know!
          </div>
        </div>

        <div className="flex  border-2 border-gray-600 h-12 rounded-lg">
          <input
            onChange={(e) => setEmail(e.target.value)}
            className="w-64 p-2 rounded-l-md"
            placeholder="Enter your E-Mail"
            type="email"
          />
          <button
            onClick={() => {
              window.scrollTo(0, 0);
              subscribe();
            }}
            className="border-0 border-l-2 border-gray-600 p-2 bg-blue-dark rounded-r-md"
            type="button"
          >
            <img src={Send} alt="Send" />
          </button>
        </div>
      </div>

      <div>
        <div className="uppercase text-xl lg:text-subtitle mt-8 lg:mt-0 lg:mb-8">
          recommended for you
        </div>

        <div className="h-96 lg:h-auto flex flex-col lg:flex-row justify-evenly lg:justify-between">
          <img
            className="cursor-pointer border-4 border-transparent border-dashed hover:border-blue-dark w-full lg:mr-8"
            src={Blog2}
            alt="blog2"
          />
          <img
            className="cursor-pointer border-4 border-transparent border-dashed hover:border-blue-dark w-full"
            src={Blog1}
            alt="blog1"
          />
        </div>
      </div>
    </div>
  );
}
