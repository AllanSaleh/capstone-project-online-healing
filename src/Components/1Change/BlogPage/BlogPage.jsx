import React from 'react';

import BlogBody from './BlogBody';
import BlogImage from './BlogImage';
import Img from './Images/BlogImg.svg';
import Recommended from './Recommended';

export default function BlogPage() {
  const BlogImg = Img;
  return (
    <>
      <BlogImage image={BlogImg} />
      <BlogBody />
      <Recommended />
    </>
  );
}
