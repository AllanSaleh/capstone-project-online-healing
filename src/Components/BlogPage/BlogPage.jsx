import React, { useState, useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import firebase from '../../firebase';

import BlogBody from './BlogBody';
import BlogImage from './BlogImage';
import Recommended from './Recommended';

const blogInfo = [
  {
    blog_id: 'Temp',
    image: 'Loading...',
    title: 'Loading...',
    intro: 'Loading...',
    subtitle: 'Loading...',
    body: ['Loading...'],
  },
];

export default function BlogPage() {
  window.scrollTo(0, 0);
  const location = useLocation();

  const [mainBlog, setMainBlog] = useState(blogInfo[0]);
  const [randomBlogs, setRandom] = useState({
    blog1: blogInfo[Math.floor(Math.random() * blogInfo.length)],
    blog2: blogInfo[Math.floor(Math.random() * blogInfo.length)],
  });

  useEffect(() => {
    firebase
      .firestore()
      .collection('articles')
      .get()
      .then((querySnapshot) => {
        const data = querySnapshot.docs.map((doc) => doc.data());
        if (location.state !== undefined)
          data.forEach((blog) => {
            if (blog.blog_id === location.state) setMainBlog(blog);
          });
        else setMainBlog(data[0]);
        setRandom({
          blog1: data[Math.floor(Math.random() * data.length)],
          blog2: data[Math.floor(Math.random() * data.length)],
        });
      });
  }, []);

  return (
    <>
      <BlogImage image={mainBlog.image} />
      <BlogBody
        title={mainBlog.title}
        intro={mainBlog.intro}
        subtitle={mainBlog.subtitle}
        body={mainBlog.body}
      />
      <Recommended blog1={randomBlogs.blog1} blog2={randomBlogs.blog2} />
    </>
  );
}
