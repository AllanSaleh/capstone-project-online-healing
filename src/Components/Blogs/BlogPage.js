import React from 'react';
import NavBar from '../NavBar/NavBar';
import Footer from '../Footer/Footer';
import BlogBody from './BlogBody';
import BlogImage from './BlogImage';
import BlogImg from './BlogImg.svg';
import Recommended from './Recommended';

export default function BlogPage() {
  return (
    <div>
      <NavBar />
      <BlogImage image={BlogImg} />
      <BlogBody />
      <Recommended />
      <Footer />
    </div>
  );
}
