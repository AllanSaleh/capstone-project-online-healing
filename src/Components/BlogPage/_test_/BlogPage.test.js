import * as React from 'react';
import * as renderer from 'react-test-renderer';
import BlogPage from '../BlogPage';

it('Blog Page renders correctly', () => {
  const tree = renderer.create(<BlogPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
