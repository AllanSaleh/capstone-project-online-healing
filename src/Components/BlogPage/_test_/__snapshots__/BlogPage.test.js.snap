// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="flex items-center lg:h-firstsection px-sides pt-24 lg:pt-navbar"
  >
    <img
      alt="Blog"
      className="w-full"
      src="BlogImg.svg"
    />
  </div>,
  <div
    className="flex flex-col justify-evenly lg:h-section px-sides my-4"
  >
    <div
      className="text-3xl lg:text-title"
    >
      Mental Health and Technology
    </div>
    <div
      className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify"
    >
      Technology and mental well-being might seem like profoundly antagonistic terms. However, there are excellent digital tools to help you reach mindfulness and to practice meditation. You can even use smartphone apps, virtual reality solutions and digital devices as a form of relieving stressing and achieving digital detox – as, after a while, you will be able to practice all types of anti-anxiety skills on your own. So, let The Medical Futurist show you how you can enhance your mental and emotional health with the help of technology.
    </div>
    <div
      className="mt-4 lg:mt-0 text-xl lg:text-subtitle"
    >
      MENTAL HEALTH AND TECHNOLOGY
    </div>
    <div
      className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify"
    >
      “Just as we observe physical hygiene to stay well, we need to cultivate a kind of emotional hygiene too,” wrote the Dalai Lama on his Twitter-feed. There are three crucial points around His Holiness’ statement. First and foremost, the popularity of health-enhancing, physically active lifestyle is soaring worldwide. Nevertheless, there’s always room for improvement: the WHO says that globally, 23 percent of adults and 81 percent of school-going adolescents are not active enough. According to the organization, adults aged 18-64 should do at least 150 minutes of moderately intense physical activity each week, which is not unaccomplishable. Moreover, the numerically and statistically driven 21st-century mind could take a great inspiration and power from the various fitness trackers and wearables.
    </div>
    <div
      className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify"
    >
      Secondly, emotional hygiene seems to be thrown into the shade nowadays. Somehow, the culture of busyness prevents us from squeezing some time for mental health into our packed schedules. Information is pouring on us from millions of communication channels; we are connecting with hundreds of ‘friends’ on social media, we are obsessed with data due to the fear of missing out and time management slices up our days into meetings and to-do-lists.
    </div>
    <div
      className="mt-4 lg:mt-0 text-lg lg:text-paragraph text-gray-700 text-justify"
    >
      The pressure to perform and the stress to achieve is the air every busy-bee is breathing. However, turbulent lifestyle comes with a price. No wonder so many people have mental health issues. In the US, that means one in five adults – more than 40 million Americans! On the old continent, 27 percent of the population has to cope with mental disorders, which means 83 million struggling Europeans. In Asia, the prevalence of depression is 20 percent in Thailand, 19.9 percent in Taiwan, 19.4 percent in Korea, 17.5 in Malaysia and 16.5 percent in China.
    </div>
  </div>,
  <div
    className="px-sides lg:h-section flex flex-col justify-around"
  >
    <div
      className="flex flex-col justify-between items-start lg:h-40"
    >
      <div>
        <div
          className="text-3xl lg:text-title leading-tight"
        >
          Sign Up For The Healing Blog
        </div>
        <div
          className="text-lg lg:text-paragraph text-gray-700"
        >
          Weekly, Ad-Free Blogs That Help You Stay In The Know!
        </div>
      </div>
      <div
        className="flex  border-2 border-gray-600 h-12 rounded-lg"
      >
        <input
          className="w-64 p-2 rounded-l-md"
          placeholder="Enter your E-Mail"
          type="email"
        />
        <button
          className="border-0 border-l-2 border-gray-600 p-2 bg-blue-dark rounded-r-md"
          type="button"
        >
          <img
            alt="Send"
            src="Send.svg"
          />
        </button>
      </div>
    </div>
    <div>
      <div
        className="uppercase text-xl lg:text-subtitle mt-8 lg:mt-0 lg:mb-8"
      >
        recommended for you
      </div>
      <div
        className="h-96 lg:h-auto flex flex-col lg:flex-row justify-evenly lg:justify-between"
      >
        <img
          alt="blog2"
          className="cursor-pointer border-4 border-transparent border-dashed hover:border-blue-dark w-full lg:mr-8"
          src="Blog2.svg"
        />
        <img
          alt="blog1"
          className="cursor-pointer border-4 border-transparent border-dashed hover:border-blue-dark w-full"
          src="Blog1.svg"
        />
      </div>
    </div>
  </div>,
]
`;
